File: cpp/Construct_the_Minimum_Bitwise_Array_I_3314.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        for (int& num: nums) {
            int ans = -1;
            if (num == 2) {
                num = -1;
                continue;
            }
            num -= (1 << __builtin_ctz(~num) - 1);
        }
        return nums;
    }
};

File: python/Construct_the_Minimum_Bitwise_Array_I_3314.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        for i, num in enumerate(nums):
            if nums[i] == 2:
                nums[i] = -1
                continue
            nums[i] -= ((num + 1) & ~num) >> 1
        return nums
